src/main/java/io/github/fabricators_of_create/porting_lib/transfer/item/ItemStorageHandler.java

```diff
@@ -50,18 +55,17 @@ public ItemStack getStackInSlot(int slot) {
 	public ItemStack insertItem(int slot, ItemStack stack, boolean sim) {
 		if(stack.isEmpty())
 			return stack;
-		ItemStack finalVal = ItemStack.EMPTY;
-		try (Transaction t = Transaction.openOuter()) {
-			long remainder = stack.getCount() - storage.insert(ItemVariant.of(stack), stack.getCount(), t);
-			if (remainder != 0) {
-				finalVal = new ItemStack(stack.getItem(), (int) remainder);
-			}
-
-			if (!sim) {
+		try (Transaction t = Transaction.openOuter()){
+			long inserted = this.storage.insert(ItemVariant.of(stack), stack.getCount(), t);
+			if (sim)
+				t.abort();
```

Comment:
manually aborting should not be needed, if a try/catch block ends it will automatically abort.